presto-main/src/main/java/com/facebook/presto/execution/CreateFunctionTask.java

```diff
@@ -93,15 +97,36 @@ private SqlInvokedFunction createSqlInvokedFunction(CreateFunction statement)
                 .setDeterminism(RoutineCharacteristics.Determinism.valueOf(statement.getCharacteristics().getDeterminism().name()))
                 .setNullCallClause(RoutineCharacteristics.NullCallClause.valueOf(statement.getCharacteristics().getNullCallClause().name()))
                 .build();
-        String body = formatSql(statement.getBody(), Optional.empty());
+        RoutineBody body = statement.getBody();
+
+        if (statement.getBody() instanceof Return) {
+            Expression bodyExpression = ((Return) statement.getBody()).getExpression();
+            Type bodyType = analysis.getType(bodyExpression);
+
+            if (!bodyType.equals(metadata.getType(returnType))) {
+                // Casting is safe-here, since we have verified that the actual type of the body is coercible to declared return type.
+                body = new Return(new Cast(bodyExpression, statement.getReturnType()));
+            }
+        }
+
+        if (statement.getBody() instanceof Return) {
+            Expression bodyExpression = ((Return) statement.getBody()).getExpression();
+            Type bodyType = analysis.getType(bodyExpression);
+
+            if (!bodyType.equals(metadata.getType(returnType))) {
+                // Casting is safe-here, since the check to verify if the two types are Coercible is verified as part of the statement analyzer step.
+                Expression castExpression = new Cast(bodyExpression, statement.getReturnType());
+                body = formatSql(new Return(castExpression), Optional.empty());
+            }
+        }
```

Comment:
Somehow you introduced duplicated code blocks. I guess that's during the squash?

Reply:
Fixed the rebase issues manually and pushed the changes.